tools/cldr-code/src/test/java/org/unicode/cldr/unittest/TestDisplayAndInputProcessor.java

```diff
@@ -392,4 +392,19 @@ public void TestBackspaceFilter() {
         String value = daip.processInput(xpath, "\btest\bTEST\b", null);
         assertEquals("Backspaces are filtered out", "testTEST", value);
     }
+
+    /**
+     * Test whether DisplayAndInputProcessor.processInput normalizes whitespace.
+     * This depends very much on the xpath, since for most xpaths NBSP is normalized to
+     * ordinary space which, if initial or final, is then removed by trim(). But for some
+     * xpaths, NBSP is retained and then the standard trim() function doesn't apply.
+     */
+    public void TestWhitespaceNormalization() {
+        DisplayAndInputProcessor daip = new DisplayAndInputProcessor(info.getEnglish(), false);
```

Comment:
I think you need a few more cases to test. Each of the 3 types of paths, and for each, samples with

A0 20
20 A0
20 A0 20
20 20 

Also, you should test this by running CLDRModify over the source and collecting the changes. That way we can make sure we don't miss some cases.